### PR TITLE
[SPARK-32497][INFRA] Installs qpdf package for CRAN check in GitHub Actions

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

CRAN check fails due to the size of the generated PDF docs as below:

```
...
 WARNING
‘qpdf’ is needed for checks on size reduction of PDFs
...
Status: 1 WARNING, 1 NOTE
See
  ‘/home/runner/work/spark/spark/R/SparkR.Rcheck/00check.log’
for details.
```

This PR proposes to install `qpdf` in GitHub Actions.

Note that I cannot reproduce in my local with the same R version so I am not documenting it for now.

Also, while I am here, I piggyback to install SparkR when the module includes `sparkr`. it is rather a followup of SPARK-32491.

### Why are the changes needed?

To fix SparkR CRAN check failure.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

GitHub Actions will test it out.

### CODE REVIEW

File: dev/run-tests.py [hunk 1]
```diff
@@ -610,8 +610,14 @@ def main():
 
     # Install SparkR
     should_only_test_modules = opts.modules is not None
-    if not should_only_test_modules:
+    test_modules = []
+    if should_only_test_modules:
+        str_test_modules = [m.strip() for m in opts.modules.split(",")]
+        test_modules = [m for m in modules.all_modules if m.name in str_test_modules]
+
+    if not should_only_test_modules or modules.sparkr in test_modules:
```
Comment (HyukjinKwon) [inline dev/run-tests.py:618]: I piggy-back this change together. We don't need to install SparkR when `--modules` is specified except that it includes `sparkr`.

**Comment by HyukjinKwon** [comment]
Okay, the SparkR build now passes successfully (https://github.com/apache/spark/runs/928060018).

**Comment by HyukjinKwon** [comment]
I will see if we really need `qpdf` at https://github.com/apache/spark/runs/928145820. If the tests pass at https://github.com/apache/spark/pull/29306/commits/772f7102a5f5ddfd62154a84ca1151518660f4fd, we can remove `qpdf`  installation and merge.

If that fails, we could just merge as is.

cc @dongjoon-hyun

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by HyukjinKwon** [comment]
Thanks @dongjoon-hyun. Merged to master.

**Comment by HyukjinKwon** [comment]
GitHub Actions build should all pass now.

**Comment by HyukjinKwon** [comment]
cc @ScrapCodes and @zhengruifeng FYI. The CRAN check _might_ fail with the error in the PR description with R 4.0.0. In that case, `qpdf` will have to be installed. I can't reproduce in my local but it fails in GitHub Actions.

**Pull request closed without merging**